FrequencyConversion/FC_solve_EoM.py

```diff
@@ -83,11 +82,11 @@ def get_prop(dw, vp, va, vc, pump, L, D=1):
     ka = 1 / va - 1 / vp
     kc = 1 / vc - 1 / vp
 
-    Ka = jnp.diag(1j * dw * ka)
-    Kc = jnp.diag(1j * dw * kc)
+    Ka = np.diag(1j * dw * ka)
+    Kc = np.diag(1j * dw * kc)
     # Bottom left and upper right (non-diagonal) blocks
 
-    J = jnp.block([[Ka, -1j * jnp.conj(D) * pump], [-1j * D * pump.conj().T, Kc]])
```

Comment:
This way of generating the interaction matrix is incorrect. By combining everything into this 'D' you are omitting a \Delta \omega term which depends on the step size used. This term comes from discretizing the integral into a sum. (see Sec.4 of https://journals.aps.org/pra/abstract/10.1103/PhysRevA.102.033519, more precisely Eq. 59a for this off diagonal term). If you also look at my function "Hprop" in "propagator.py" I have a very convoluted way of expressing that \Delta \omega.